### PR TITLE
make coordinator db polling for list of segments more robust

### PR DESCRIPTION
**Pull request by himanshug**
if any single record in segments table is corrupted then coordinator fails to load the list of segments from db and in turn stops making any progress on assigning new segment to historicals
we had an outage today due to this not completing handoffs and realtime pipeline getting stuck.

this PR changes the behavior to make coordinator more robust to continue making progress in such event.

### CODE REVIEW

**Comment by fjy** [comment]
:+1:

File: server/src/main/java/io/druid/metadata/SQLMetadataSegmentManager.java [hunk 1]
```diff
@@ -466,6 +469,19 @@ public DataSegment map(int index, ResultSet r, StatementContext ctx)
         return;
       }
 
+      segments = new ArrayList<>(
+          Collections2.filter(
+              segments, new Predicate<DataSegment>()
```
Comment (xvrl) [inline server/src/main/java/io/druid/metadata/SQLMetadataSegmentManager.java]: you can simply use `Predicates.notNull()`

File: server/src/main/java/io/druid/metadata/SQLMetadataSegmentManager.java [hunk 2]
```diff
@@ -466,6 +469,19 @@ public DataSegment map(int index, ResultSet r, StatementContext ctx)
         return;
       }
 
+      segments = new ArrayList<>(
```
Comment (xvrl) [inline server/src/main/java/io/druid/metadata/SQLMetadataSegmentManager.java]: any reason we need to copy the list here?

**Comment by xvrl** [comment]
@himanshug just a few minor comments for conciseness, :+1: once those are addressed.

**Comment by himanshug** [comment]
@xvrl made updates to address all comments.

**Pull request merged**